docs/source/deployment-considerations.rst

```diff
@@ -0,0 +1,91 @@
+Deployment Considerations
+=========================
+
+Thanks to the efforts of the open-source community, there are tools to deploy Dask :ref:`pretty much anywhere <deployment-options>`—if you can get computers to talk to each other, you can probably turn them into a Dask cluster.
+
+**However, getting Dask running is often not the last step, but the first step.** This document attempts to cover some of the things *outside of Dask* you may have to think about when managing a Dask deployment.
+
+These challenges are especially relevant when managing Dask for a team or organization, or transitioning into Dask for production (automated) use, but many will also come up for individual Dask users on distributed systems.
+
+
+Consistent software environments
+--------------------------------
+For Dask to function properly, the same set of Python packages, at the same versions, need to be installed on the scheduler and workers as on the client. One of the most common stumbling points in deploying Dask on multiple machines is keeping what's installed on the cluster up to date with what's on the client, especially if they run on different systems (a laptop and a cloud provider, for example). For possible approaches, see :doc:`how-to/manage-environments`.
+
+Some ways of maintaining consistent environments may also require extra infrastructure. For example, using Docker images is common in cloud deployments, but then you need somewhere to store the images, such as `DockerHub <https://hub.docker.com/>`_, `AWS ECR <https://aws.amazon.com/ecr/>`_, `GCP Container Registry <https://cloud.google.com/container-registry>`_, etc. As use matures, you'll also want to version-control your Dockerfile and automatically build and publish new images when it changes, using a CI/CD system like `GitHub Actions <https://github.com/marketplace/actions/build-and-push-docker-images>`_, `Google Cloud Build <https://cloud.google.com/build/docs/build-push-docker-image>`_, or others.
+
+Software environments can be particularly challenging when managing Dask for multiple users. Sometimes, using the same locked-down set of packages is sufficient for a team, but often individuals will need different packages, or different versions of the same package (including Dask itself!), to be productive. In these cases, you might end up giving end-users access to the build system (requiring them to build and maintain Docker images themselves, for example), or if that isn't allowable in your organization, creating custom infrastructure, or resorting to workarounds like the :class:`~distributed.diagnostics.plugin.PipInstall` or :class:`~distributed.diagnostics.plugin.UploadDirectory` plugins.
+
+Additional challenges can include getting local packages or scripts onto the cluster (and ensuring they're up to date), as well as packages installed from private Git or PyPI repos.
+
+
+Observability
```

Comment:
This is super important, but I feel like this is one of the last things folks think about. I would probably move this below other sections like cost and credentials.

Reply:
That was actually why I moved it up here; I know people don't usually think about it up front, but I wanted to make it more prominent since it's so important.

Also note that I mixed log retention in with metrics. Maybe those are worth splitting; I think log retention should be quite high (you're really not going to have a good time if you don't even keep logs around), but metrics usually come a bit later in your deployment journey.